app/src/main/java/com/example/stromtracker/ui/geraete/GeraeteListAdapter.kt

```diff
@@ -33,14 +37,25 @@ class GeraeteListAdapter(
         return geraeteList.size
     }
 
-
     override fun onBindViewHolder(holder: GeraeteListAdapter.GeraeteViewHolder, position: Int) {
         holder.mTextView.text = geraeteList[position].getName()
         //.withSign(1) lässt den "Verbrauch" bzw. die Produktion von Produzenten positiv anzeigen, da diese als negativer Verbrauch in der DB gespeichert ist
         holder.mVerbrauchView.text =
-            geraeteList[position].getJahresverbrauch().withSign(1).toString()
-        holder.mRaumView.text = raumList[geraeteList[position].getRaumID() - 1].getName()
-
+            String.format("%.2f", geraeteList[position].getJahresverbrauch().withSign(1))
+        val raumID = geraeteList[position].getRaumID()
+        for (raum in raumListHaushalt) {
+            if (raum.getRaumID() == raumID) {
+                holder.mRaumView.text = raum.getName()
+                break
+            }
+        }
```

Comment:
Das könnte man ggf. auch in eine Funktion Auslagern, und dann beim sortieren nach den namen wieder verwenden oder?

Reply:
Ich weiß nicht, ob eine for Schleife innerhalb einer Sortierungsschleife sinnvoll ist, wie im Kommentar geschrieben.

In eine Funktion auslagern macht für mich nicht viel Sinn, es ist ja nur eine for Schleife.